copulas/univariate/selection.py

```diff
@@ -27,7 +27,7 @@ def select_univariate(X, candidates):
             if ks < best_ks:
                 best_ks = ks
                 best_model = model
-        except ValueError:
+        except Exception:
```

Comment:
Do we want to swallow all exceptions? I'm wondering if this would cause us to miss other potential errors in this function.

Maybe we could match on the exception error string we expect and raise the exception in all other cases.

Reply:
Well, that was precisely the main point of the change :-)
Scipy made changes that now make some fit errors show up as plain `Exception`, so we cannot narrow it any more than that.
When it comes to filtering by the Exception message, it would be a very fragile approach, because we could end up having to search for more than one message. think about other "known" Exceptions that we have not seen yet, but also think about the possibility of having message variations in future `scipy` releases.

But most importantly: Do we really want to let other exceptions go? I think we do not. If there is anything that makes a `candidate` crash while other candidates can be fitted without issues, we do not want the selection process to crash because of that one. We rather want the selection to discard the candidate and move on with the ones that work.

If anything, we could add an `if` block at the end that raises an explicit error message if ALL the candidates failed, but I would rather open a separate issue and PR for that, because it would not really be related to the issue that is being fixed on this PR.

Reply:
I see. I guess my next question is, do we want to be aware of the errors being thrown from candidates, even if we do not raise them? If so, we could potentially log them

Reply:
Good point @katxiao More than logging I would use a `warning`, so the user has the ability to either add them to logs or ignore them if the thing becomes to verbose. I'll add that on the new issue specification.